core/renderer/backend/ProgramState.cpp

```diff
@@ -127,6 +127,22 @@ void TextureInfo::assign(TextureInfo&& other)
     }
 }
 
+void TextureInfo::assign(int slot, int index, backend::TextureBackend* texture)
+{
+    if (textures.size() != 1 or textures[0] != texture or slots[0] != slot or indexs[0] != index)
+    {
+        releaseTextures();
```

Comment:
should invoke indexs.clear(); slots.clear() in releaseTextures if you use xxx.resize(1, val), otherwise the val can't be update